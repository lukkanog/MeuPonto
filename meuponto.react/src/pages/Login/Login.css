*{
    padding: 0;
    margin: 0;
    font-family: 'Century Gothic', sans-serif, cursive;
}


.Login{
    display: flex;
    justify-content: space-between;
    background-image: url("../../assets/img/background-login.png");
    background-repeat: repeat;   
    background-position: right;
    /* background-position: center; */
    background-size: contain;
    height: 100vh;
}

.purple-half{
    clip-path: polygon(0 0, 91% 0, 24% 100%, 0% 100%);
    width: 95%;
    background-color: #5E0A6E;
    display: flex;
    flex-flow: column nowrap;
    justify-content: space-between;
    padding: 2em 3em;
}

.form-half{
    /* width: 85%; */
    display: flex;
    flex-flow: column-reverse;
    justify-content: center;
    align-content: center;
}

textarea:focus, input:focus, select:focus{  
    outline: #5E0A6E solid 1px;
    box-shadow: 2px 1px 5px #5E0A6E;
  }


.form-box{
    height: 40%;
    width: 30%;
    background-color: #FFF; 
    position: relative;
    top: 32.5%;
    right: 25vh;
    border: 1px solid #260d6042;
    border-radius: 10px;
    box-shadow: 15px 7px 10px #260d6041 ;
    padding: 1em 0;
    
    display: flex;
    flex-flow: column nowrap;
    align-items: center;
    justify-content: space-around;
}


.form-box h1{
    color: #5E0A6E;
    font-weight: 500;
    font-size: 2em;
    margin-bottom: -1em;
}

.login-form{
    display: flex;
    flex-flow: column nowrap;
    justify-content: space-around;
    /* height: 10%; */
    width: 80%;
}

.input-text-login{
    padding: 1.2em 0.5em;
    height: 1.5em;
    border: 1px solid #A69797;
    border-radius: 8px;
    margin: 0.5em 0;
}

.submit{
    color: #FFF;
    font-size: 1.2em;
    background-color: #5E0A6E;
    border: 0.1px solid  #5E0A6E;
    border-radius: 7px;
    padding: 0.5em;
    margin-top: 0.7em;
    transition: ease-in-out 0.2s
}

.submit:hover{
    box-shadow: 2px 2px 10px #5E0A6E;
}


.logo-principal{
    position: relative;
    top: 30%;
    left: 12.5%;
    height: 27.5%;
    width: 40%;
}


.logo-senai{
    width: 20%;
    height: 5.5vh;
    /* margin-left: 1em; */
}

.error-alert{
    min-height: 20vh;
    margin: auto;
}

