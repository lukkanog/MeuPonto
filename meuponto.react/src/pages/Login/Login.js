import React, { Component } from "react";
import "./Login.css";
import logoMeuPonto from "../../assets/img/LogoMeuPonto.png";
import logoSenai from "../../assets/img/senai-logo.png";
import Modal from "react-responsive-modal";
import ErrorModal from "../../components/ErrorModal";
import Axios from "axios";



export default class Login extends Component {
    constructor(props) {
        super(props);
        this.state = {
            email: null,
            senha: null,
            erro: false,
            mensagemErro: "Ocorreu um erro inesperado",
            disabled: true,
        }
    }

    atualizarEstado = (event) => {
        event.preventDefault();
        this.setState({ [event.target.name]: event.target.value });
    }


    tentarFazerLogin = (event) => {
        event.preventDefault();

        if (this.state.email === null || this.state.email === "") {
            this.setState({ erro: true });
            this.setState({ mensagemErro: "Preencha o campo de email corretamente" });
        } else if (this.state.senha === null || this.state.senha === "") {
            this.setState({ erro: true });
            this.setState({ mensagemErro: "Preencha o campo da senha corretamente" });
        } else {
            this.efetuarLogin();
        }
    }

    fecharModal = () => {
        this.setState({ erro: false });
    }

    efetuarLogin = () => {
        try {

            var requestBody = {
                email: this.state.email,
                senha: this.state.senha
            }

            let requestUrl = "http://192.168.5.98:5000/api/login"

            Axios.post(requestUrl, requestBody)
                .then(response => {
                    console.log(response);

                    if (response.status === 200) {
                        localStorage.setItem("usuario-meuPonto", response.data.token);
                        this.props.history.push("/")
                    }
                    else if (response.status === 404) {
                        this.setState({ erro: true });
                        this.setState({ mensagemErro: "Email e/ou senha incorretos" })
                    } else {
                        this.setState({ erro: true });
                        this.setState({ mensagemErro: "Ocorreu um erro inesperado. Descrição do erro: " + response.data })
                    }
                })
                .catch(error => {
                    console.log(error)
                    this.setState({ erro: true });
                    this.setState({ mensagemErro: "Ocorreu um erro inesperado. Descrição do erro: " + error })
                })

        } catch (error) {
            this.setState({ erro: true });
            this.setState({ mensagemErro: "Ocorreu um erro inesperado. Por favor, tente novamente" });
        }
    }


    render() {
        return (
            <div className="Login">
                <div className="purple-half">
                    <img className="logo-principal" src={logoMeuPonto} alt="Logo meu ponto" />
                    <img className="logo-senai" src={logoSenai} alt="Logo do SENAI"/>
                </div>

                <Modal
                    className="error-alert"
                    open={this.state.erro}
                    onClose={this.fecharModal}
                    focusTrapped={false}

                    children={<ErrorModal mensagemErro={this.state.mensagemErro} />}
                />

                {/* <div className="form-half"> */}
                <div className="form-box">

                    <h1>Login</h1>

                    <form className="login-form" onSubmit={this.tentarFazerLogin}>
                        <input
                            onInput={this.atualizarEstado}
                            className="input-text-login"
                            name="email"
                            type="email"
                            placeholder="Email"
                            minLength="3"
                            maxLength="100"
                        />

                        <input
                            onInput={this.atualizarEstado}
                            className="input-text-login"
                            name="senha"
                            type="password"
                            placeholder="Senha"
                            minLength="8"
                            maxLength="80"
                        />

                        <input
                            className="submit"
                            type="submit"
                            value="Entrar"
                        />
                    </form>

                </div>
            </div>
            // </div>
        )

    }

}