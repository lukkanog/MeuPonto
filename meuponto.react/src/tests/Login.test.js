import React,{Component} from "react";
import renderer from "react-test-renderer";
import Login from "../pages/Login/Login";

test("Compara tela de login ao snapshot",() =>{
    let render = renderer.create(<Login />);
    let tree = render.toJSON();
    expect(tree).toMatchSnapshot();
})