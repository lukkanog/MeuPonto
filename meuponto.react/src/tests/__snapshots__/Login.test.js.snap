// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compara tela de login ao snapshot 1`] = `
<div
  className="Login"
>
  <div
    className="purple-half"
  >
    <img
      className="logo-principal"
      src="LogoMeuPonto.png"
    />
    <img
      className="logo-senai"
      src="senai-logo.png"
    />
  </div>
  <div
    className="form-box"
  >
    <h1>
      Login
    </h1>
    <form
      className="login-form"
      onSubmit={[Function]}
    >
      <input
        className="input-text-login"
        maxLength="100"
        minLength="3"
        name="email"
        onInput={[Function]}
        placeholder="Email"
        type="email"
      />
      <input
        className="input-text-login"
        maxLength="80"
        minLength="8"
        name="senha"
        onInput={[Function]}
        placeholder="Senha"
        type="password"
      />
      <input
        className="submit"
        type="submit"
        value="Entrar"
      />
    </form>
  </div>
</div>
`;
